ur_robot_driver/include/ur_robot_driver/comm/producer.h

```diff
@@ -90,8 +99,10 @@ class URProducer : public IProducer<HeaderT>
    *
    * \returns Success of reading and parsing the package
    */
-  bool tryGet(std::vector<std::unique_ptr<URPackage<HeaderT>>>& products)
+  bool tryGet(std::vector<std::unique_ptr<URPackage<HeaderT>>>& products) override
   {
+    // TODO This function has become really ugly! That should be refactored!
+
```

Comment:
Thinking about it more I don't think it has actually become worse, but some refactoring (or at least adding some comments) still seems like a good idea.

Reply:
Yes, it was worse in between... But I'd like to leave this comment there for the moment.